plugins/woocommerce-blocks/tests/e2e/tests/product-collection/product-collection.block_theme.spec.ts

```diff
@@ -741,4 +743,91 @@ test.describe( 'Product Collection', () => {
 			);
 		} );
 	} );
+
+	test.describe( 'Location is recognised correctly', () => {
+		const getProductIdFromResponse = async ( response: Response ) => {
+			const products = await response.json();
+			return products[ 0 ].id;
+		};
+		const getLocationDetailsFromRequest = ( request: Request ) => {
+			const searchParams = new URLSearchParams( request.url() );
+			return {
+				locationType: searchParams.get( 'location[type]' ),
+				productId: searchParams.get(
+					'location[sourceData][productId]'
+				),
+			};
+		};
+		const verifyProductId = async ( page: Page, productSlug: string ) => {
+			const productResponsePromise = page.waitForResponse(
+				( response: Response ) => {
+					const url = response.url();
+					return (
+						url.includes( 'wp/v2/product' ) &&
+						url.includes( `slug=${ productSlug }` )
+					);
+				}
+			);
+			const locationRequestPromise = page.waitForRequest(
+				( request: Request ) => {
+					const url = request.url();
+					return (
+						url.includes( 'wp/v2/product' ) &&
+						url.includes( 'isProductCollectionBlock=true' )
+					);
+				}
+			);
+			const [ productResponse, locationRequest ] = await Promise.all( [
+				productResponsePromise,
+				locationRequestPromise,
+			] );
+			const expectedProductId = await getProductIdFromResponse(
+				productResponse
+			);
+			const { locationType, productId } =
+				getLocationDetailsFromRequest( locationRequest );
+
+			expect( locationType ).toBe( 'product' );
+			expect( productId ).toBe( String( expectedProductId ) );
+		};
+
+		test( 'in specific Single Product template', async ( {
+			page,
+			pageObject,
+			editorUtils,
+		} ) => {
+			const productName = 'Cap';
+			const productSlug = 'cap';
+
+			await editorUtils.openSpecificProductTemplate(
+				productName,
+				productSlug
+			);
+
+			await editorUtils.insertBlockUsingGlobalInserter(
+				'Product Collection (Beta)'
+			);
+			await pageObject.chooseCollectionInTemplate( 'featured' );
+
+			await verifyProductId( page, productSlug );
```

Comment:
The `waitForRequest` calls inside `verifyProductId` might not work in a stable manner because they might be already over when `verifyProductId` is called. I'd recommend initiating the request promise before the `chooseCollectionInTemplate` is called and resolving once it's done:

```suggestion
			const verifyProductIdPromise = verifyProductId( page, productSlug );
			await pageObject.chooseCollectionInTemplate( 'featured' );

			await verifyProductIdPromise;
```